lib/heap_sort.rb

```diff
@@ -1,8 +1,19 @@
-
-
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
+
 def heapsort(list)
```

Comment:
You are adding `n` items to a heap and each add takes `log n` time.  Then you remove with the same time complexity.  

So your time complexity is `O(n log n)`, which is as fast as you can sort a generic list.